javascript-vanilla-v2/main.js

```diff
@@ -0,0 +1,158 @@
+
+      const jsonTextBoxEl = document.getElementById('json-text-box')
+      if (jsonTextBox) {
+        jsonTextBoxEl.innerHTML = JSON.stringify(
+          flagsJson,
+          '',
+          indentValue.indent,
+        )
+        jsonTextBoxEl.classList.remove('hidden')
+      } else {
+        jsonTextBoxEl.classList.add('hidden')
+      }
+
+      const buttonEl = document.getElementById('button')
+      if (button) {
+        buttonEl.classList.remove('hidden')
+      } else {
+        buttonEl.classList.add('hidden')
+      }
+
+      const colouredButtonEl = document.getElementById('coloured-button')
+      if (colouredButton) {
+        colouredButtonEl.classList.remove('hidden')
+        colouredButtonEl.style.backgroundColor = colouredButtonValue
+      } else {
+        colouredButtonEl.classList.add('hidden')
+      }
+
+      const featuresTableEl = document.getElementById('features')
+      const listEl = document.getElementById('list')
+      if (fontSize) {
+        featuresTableEl.style.fontSize = fontSizeValue + 'px'
+        listEl.style.fontSize = fontSizeValue + 'px'
+      } else {
+        featuresTableEl.style.fontSize = '16px'
+        listEl.style.fontSize = '16px'
+      }
+
+      const showIconsEl = document.getElementById('icons')
+      if (showIcons) {
+        showIconsEl.classList.remove('hidden')
+      } else {
+        showIconsEl.classList.add('hidden')
+      }
+
+      const announcementEl = document.getElementById('announcement')
+      if (announcement) {
+        announcementEl.classList.remove('hidden')
+        announcementEl.innerHTML = announcementValue
+      } else {
+        announcementEl.classList.add('hidden')
+      }
+
+      const imagesEl = document.getElementById('images')
+      if (images) {
+        imagesEl.classList.remove('hidden')
+      } else {
+        imagesEl.classList.add('hidden')
+      }
+
+      const loginButton = document.getElementById('js-login')
+      const logoutButton = document.getElementById('js-logout')
+
+      if (flagsmith.identity) {
+        loginButton.classList.add('hidden')
+        logoutButton.classList.remove('hidden')
+      } else {
+        loginButton.classList.remove('hidden')
+        logoutButton.classList.add('hidden')
+      }
+    }
+    // To enable debug mode, set debug to 1 on line 3.
+    if (debug) {
+      console.log('is_Server', isFromServer)
+      console.log('allFlags', flagsJson)
+      console.log('json_textbox', jsonTextBox, indentValue)
+      console.log('font_size', fontSize, fontSizeValue)
+      console.log('announcement', announcement, announcementValue)
+      console.log('new_button', button)
+      console.log('coloured_button', colouredButton, colouredButtonValue)
+    }
+  },
+})
+
+// Set as a identity if you want to fetch flags only for this specific identity.
+// identity: 'some_identity',
+// Set as a trait if you want to fetch flags only for this specific trait.
+// traits: 'some_trait',
```

Comment:
Shouldn't this be part of `flagsmith.init` ?

Reply:
That's right. I put it down there while i was writing the other topics. Gonna change it and put it right before "onChange".

Reply:
Since it is a more general example, I don't think it should be? Notice what we are doing with the login/logout and identity and segment overrides.

Reply:
@gnumoreno it doesn't make sense outside of the `init` call. Looks like it's been moved there though so I'll resolve this conversation thread.